pkg/operator/sync.go

```diff
@@ -353,8 +353,12 @@ func getIgnitionHost(infraStatus *configv1.InfrastructureStatus) (string, error)
 		case configv1.OvirtPlatformType:
 			ignitionHost = net.JoinHostPort(infraStatus.PlatformStatus.Ovirt.APIServerInternalIP, securePortStr)
 		case configv1.VSpherePlatformType:
-			if infraStatus.PlatformStatus.VSphere.APIServerInternalIP != "" {
-				ignitionHost = net.JoinHostPort(infraStatus.PlatformStatus.VSphere.APIServerInternalIP, securePortStr)
+			if infraStatus.PlatformStatus.VSphere != nil {
```

Comment:
Indeed we haven't changed this code in a while, so wondering why we are hitting this bug now. My main concern is, is this something expected on vSphere to have infraStatus.PlatformStatus.VSphere nil or something is missing during vSphere cluster install. Also, can it cause any possible issue later with cluster where we are not adding securePortStr on a vSphere cluster?

Reply:
Ah I think this was due to the recent PR merge to manage user data secret, as a result of

https://github.com/openshift/machine-config-operator/commit/129d6e4ba020b50028ba28d51e4289cbc3462e2f#diff-3e205577bec1a1d1711df8bfeff30e63f044b24cfbdc69bbd7d0052fdc881891

and 

https://github.com/openshift/machine-config-operator/commit/59f1381ec54d0ccb540b636feef5bbfe8095caeb#diff-3e205577bec1a1d1711df8bfeff30e63f044b24cfbdc69bbd7d0052fdc881891

Which we didn't do in sync.go prior to this since we (presumably) didn't need to figure out the ignition port. I may have been a little hasty on merging that PR. Maybe we can revisit that whole section? Otherwise the fix here (although it would get us past the failure) will still cause problems for the managed secret.

Reply:
It looks like we may hit this bug on other platform handled in getIgnitionHost() right?
Since you have better context of https://github.com/openshift/machine-config-operator/pull/2827 , do you think it is better to revert the PR or check with PR author to see if we can fix the issue?

Reply:
cc @zaneb 

I think this is because some of the old ported code may not be updated with updated assumptions on what can be nil on what platform.

Reply:
There's no immediate plans to use the managed secret in vSphere, so I think we should go ahead with this fix.

It's not clear to me why infraStatus.PlatformStatus would be non-nil but infraStatus.PlatformStatus.VSphere nil on a vSphere cluster.

Reply:
Ok so I dug around for this, and we aren't exactly internally consistent on this either. Namely, we have 2 functions that query for this field,

1. in the operator: https://github.com/openshift/machine-config-operator/blob/b50080b6bdc344496dc001bf7f0b25fce5d039d6/pkg/operator/render.go#L272 - this doesn't actually seem to have that exact detection
2. in the template controller: https://github.com/openshift/machine-config-operator/blob/aaa600e951b3a76062c0a5acca0abd06da4e55b4/pkg/controller/template/render.go#L466 - This does have the extra detection and the reasoning.

I wonder why the operator one doesn't error. In any case I think we should probably look to de-duplicate the above 2 functions and then use the correct one here. At least we should be safe for the other platforms, so it's ok either way.

We can just merge this PR as is to unblock CI -> properly de-dupe as well.